include/SFML/Audio/InputSoundFile.hpp

```diff
@@ -206,7 +206,7 @@ class SFML_AUDIO_API InputSoundFile
     /// \brief Deleter for input streams that only conditionally deletes
     ///
     ////////////////////////////////////////////////////////////
-    struct StreamDeleter
+    struct SFML_AUDIO_API StreamDeleter
```

Comment:
Exporting private helper does seem right :thinking:

Reply:
Seems like it fixed the issue. Can't understand why though.

Reply:
Not sure but I guess that a compiler-generated destructor can be inlined. The compiler-generated implementation inlined in client code could not find StreamDeleter.
A cleaner solution IMHO is to declare the destructor and provide a default implementation in cpp file as before. To enable move semantics, declaring default move constructor and move assignment operator should work.

Reply:
> To enable move semantics, declaring default move constructor and move assignment operator should work.

The whole point of this PR was to avoid needing to do that though.

Reply:
On second thought, exported classes have their private methods visible to the linker. With that in mind, it make sense to have this helper class exported too. It just felt weird to see it explicitly at first.